jobs/db_store.go

```diff
@@ -127,6 +127,18 @@ func (s *DatabaseStore) PutJob(inst flux.InstanceID, job Job) (JobID, error) {
 	}
 
 	err = s.Transaction(func(s *DatabaseStore) error {
+		if job.Key != "" {
+			var count int
+			err := s.conn.QueryRow(`
+			SELECT count(*) FROM jobs WHERE key = $1 AND claimed_at IS NULL
```

Comment:
I initially had `finished_at IS NULL`, but that precludes re-scheduling the current job. (You have to mark it finished prematurely, which is a bit weird). Hence I went with `claimed_at` instead.